book/src/ui/simple_button.md

```diff
@@ -0,0 +1,111 @@
+# Build your own!
+
+In this chapter we will guide you through building your own button in Amethyst!
+
+### Bulding blocks
+
+The components you can use in order to build your button are as goes: 
+
+- [UiTransform](https://docs.amethyst.rs/master/amethyst_ui/struct.UiTransform.html) - 
+used for positioning your button on the screen (same as Transform but for the UI elements)
+
+- [UiText](https://docs.amethyst.rs/master/amethyst_ui/struct.UiText.html) - 
+if you want your button to have any text displayed
+
+- [UiImage](https://docs.amethyst.rs/master/amethyst_ui/enum.UiImage.html) - 
+if you want your button to display a texture 
+
+
+You don't have to use all three at the same time of course but variations of two (`UiTransfrom` is always needed!).
+
+### Creating the `UiTransform`
+
+One way of defining a `UiTransform` is like so: 
+
+```rust 
+let ui_transfrom = UiTransform::new(
+    String::from("simple_button"), // id
+	Anchor::Middle, 			   // anchor 
+	Anchor::Middle, 			   // pivot 
+	0f32, 						   // x 
+	0f32, 						   // y 
+	0f32,						   // z 
+	100f32,						   // width
+	30f32, 						   // height
+);
+```
+
+The `id` field of the transform is basically like the name. You can use this in combination with the
+[UiFinder](https://docs.amethyst.rs/master/amethyst_ui/struct.UiFinder.html) to fetch the transfroms through the systems.
+
+Assuming the entity has no parent, whatever is set as the `anchor` field will be placed relative to the screen. In our case
+we set it to `Anchor::Middle` and it will be drawn in the middle of the screen. The `pivot` field will center the widget
+relative to itself - this in turn is the reason why our `x` and `y` fields are `0f32`. The `z` field of this struct 
+is used for "depth" ordering of the ui elements.
+
+The `width` and `height` fields are also important. They represent the area that will register the events like hovering over 
+with the mouse, clicking and dragging. If you built the entity with the `UiText` component this also determines if the text will be rendered, 
+meaning you need 
+to set the area big enough for the text to fit in!
+
+
+### Creating the `UiText` 
+
+```rust
+let ui_text = UiText::new(
+	font_handle, 				   // font
+    String::from("Simple Button"), // text
+    font_size: 25f32, 			   // font_size
+    color: [1.0, 1.0, 1.0, 0.5],   // color
```

Comment:
Unless this is some crazy new Rust syntax, it looks as if it won't work.

Reply:
hm, it should work, it's a fixed sized array with the values specified;
`25f32` is also something that's been around for a while ([literal suffixes](https://doc.rust-lang.org/stable/rust-by-example/types/literals.html))

Reply:
He might be refering to the argument names that i forgot to remove. :)

Reply:
Indeed. I actually spent some minutes googling whether some keyword argument RFC had been accepted/merged in Rust 😆 And there is an actual proposal which looks very similar.

Reply:
BTW, thanks for this PR. I am getting familiar with Amethyst's UI these days and this has been quite helpful.